osu.Game/Skinning/Editor/SkinBlueprintContainer.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Testing;
+using osu.Game.Rulesets.Edit;
+using osu.Game.Screens.Edit.Compose.Components;
+using osu.Game.Screens.Play.HUD;
+
+namespace osu.Game.Skinning.Editor
+{
+    public class SkinBlueprintContainer : BlueprintContainer<ISkinnableComponent>
+    {
+        private readonly Drawable target;
+
+        public SkinBlueprintContainer(Drawable target)
+        {
+            this.target = target;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            checkForComponents();
+        }
+
+        private void checkForComponents()
+        {
+            foreach (var c in target.ChildrenOfType<ISkinnableComponent>().ToArray()) AddBlueprintFor(c);
+
+            Scheduler.AddDelayed(checkForComponents, 1000);
```

Comment:
This seems a bit off. `ChildrenOfType()` usage in non-test code every second sounds like a potential performance problem. Probably not something for right now but yeah.

Reply:
The reason for it running more than once is because some component late-load. But I wrote this with the intention of leaving it into the foreseeable future.

Open to proposals on how we can improve on this going forward without adding complexity.

Reply:
I see, that does complicate matters. I don't think I see any good option of eliminating this. Could have the skinnables register themselves on load complete, but that just exchanges computational overhead for implementation overhead. May be fine to leave then...